source/server/listener_manager_impl.cc

```diff
@@ -128,11 +128,6 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, const std::st
       workers_started_(workers_started), hash_(hash),
       local_drain_manager_(parent.factory_.createDrainManager(config.drain_type())),
       config_(config), version_info_(version_info) {
-  if (config.filter_chains().empty()) {
```

Comment:
I don't understand why this is removed. Is this handled elsewhere now, or we just want to let the counter for no matching filter chain go up, or is there some different semantic in place now for this case?

Reply:
This is enforced by proto constraint already, so the check was useless:
```
  repeated listener.FilterChain filter_chains = 3
      [(validate.rules).repeated .min_items = 1, (gogoproto.nullable) = false];
```